### DISCUSSION TITLE
Dynamic Backend endpoint generation

### DISCUSSION THREAD

**Discussion by RagunathInGIT** [Kong/kong#7627] [Help]
We have a requirement where we need to dynamically generate the backend service endpoint based on the incoming URI from consumer. Any suggestion on this? Ours is Kong running on K8s with postgres DB.

**Comment by kikito** [comment]
Could you please add more details to your question? An example of the type of URI and the resulting endpoints might help. Is the list of possible services finite and known?

**Comment by jeremyjpj0916** [comment]
We did something similar where URI helped dictate hostname we routed too, some fork of this kinda plugin could work for yah.  https://github.com/Optum/kong-path-based-routing
